### ISSUE TITLE
RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.

### ISSUE THREAD

**Issue by 2bndy5** [rst2pdf/rst2pdf#980]
I get the the following warning when using rst2pdf with sphinx
>RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.

### the entire output
```text
c:\users\%USER_NAME%\appdata\local\programs\python\python38-32\lib\site-packages\rst2pdf\pdfbuilder.py:232: RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.
  t = copy(self.env.indexentries)
c:\users\%USER_NAME%\appdata\local\programs\python\python38-32\lib\site-packages\rst2pdf\pdfbuilder.py:235: RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.
  docname: self.env.indexentries[docname + '-gen']
c:\users\%USER_NAME%\appdata\local\programs\python\python38-32\lib\site-packages\rst2pdf\pdfbuilder.py:238: RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.
  self.env.indexentries = {}
c:\users\%USER_NAME%\appdata\local\programs\python\python38-32\lib\site-packages\rst2pdf\pdfbuilder.py:240: RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.
  self.env.indexentries[dname] = t.get(dname, [])
c:\users\%USER_NAME%\appdata\local\programs\python\python38-32\lib\site-packages\rst2pdf\pdfbuilder.py:242: RemovedInSphinx40Warning: env.indexentries() is deprecated. Please use IndexDomain instead.
  self.env.indexentries = t
```
### Versions
- Python 3.8.2 (3.8.5 on WSL ubuntu)
- rst2pdf 0.98
- reportlab 3.5.59
- Sphinx 3.0.3 (3.4.3 on WSL ubuntu)

this warning output is consistent using Windows 10 and WSL ubuntu (with version listed above)

**Comment by lornajane** [comment]
This could be a Sphinx version incompatibility problem - rst2pdf is pinned to sphinx `1.8.5` although #984 is open as we would really like to update it. I know there are breaking changes between here and the `3.x` versions, but I'm not sure if that's what's happening here.

**Comment by 2bndy5** [comment]
Turns out this a pre-emptive warning about intended changes to Sphinx v4.0 (not released as of this writing). I found [the source code producing this warning](https://github.com/sphinx-doc/sphinx/blob/b51c947c518fc96b9fe0a59e502e67ab5334e320/sphinx/environment/__init__.py#L645-L659) (if it helps for reference).
```python
    @property
    def indexentries(self) -> Dict[str, List[Tuple[str, str, str, str, str]]]:
        warnings.warn('env.indexentries() is deprecated. Please use IndexDomain instead.',
                      RemovedInSphinx40Warning, stacklevel=2)
        from sphinx.domains.index import IndexDomain
        domain = cast(IndexDomain, self.get_domain('index'))
        return domain.entries

    @indexentries.setter
    def indexentries(self, entries: Dict[str, List[Tuple[str, str, str, str, str]]]) -> None:
        warnings.warn('env.indexentries() is deprecated. Please use IndexDomain instead.',
                      RemovedInSphinx40Warning, stacklevel=2)
        from sphinx.domains.index import IndexDomain
        domain = cast(IndexDomain, self.get_domain('index'))
        domain.data['entries'] = entries
```

**Comment by akrabat** [comment]
Fixed when #1020 is merged..
